code/autoannotate.py

```diff
@@ -0,0 +1,321 @@
+
+        # now add match to new_taxa
+        if match == 'species':
+            species_match += 1
+
+            if len(species_set) > 1:
+                species = choose_taxonomy(name, species_set, ref_taxa_fp)
+            else:
+                species = list(species_set)[0]
+
+            if species not in new_taxa.keys():
+                new_taxa[species] = (name, t[1])
+            else:
+                # if species is replicated, collapse abundances
+                new_taxa[species] = (name,
+                                     add_lists(new_taxa[species][1], t[1]))
+                duplicates.append((name, species))
+
+        elif match == 'genus':
+            genus_match += 1
+
+            if len(genus_set) > 1:
+                genus = choose_taxonomy(name, genus_set, ref_taxa_fp)
+            else:
+                genus = list(genus_set)[0]
+
+            if genus not in new_taxa.keys():
+                new_taxa[genus] = (name, t[1])
+            else:
+                # if genus is replicated, collapse abundances
+                new_taxa[genus] = (name, add_lists(new_taxa[genus][1], t[1]))
+                duplicates.append((name, genus))
+
+        elif match == 'family':
+            family_match += 1
+
+            if family not in new_taxa.keys():
+                new_taxa[family] = (name, t[1])
+            else:
+                # if genus is replicated, collapse abundances
+                new_taxa[family] = (name, add_lists(new_taxa[family][1], t[1]))
+                duplicates.append((name, family))
+
+        # if failed, user needs to manually search and input new string
+        else:
+            no_match += 1
+
+            lineage = manual_search(name, ref_taxa_fp)
+            if lineage not in new_taxa.keys():
+                new_taxa[lineage] = (name, t[1])
+            else:
+                # if genus is replicated, collapse abundances
+                new_taxa[lineage] = (name, add_lists(new_taxa[lineage][1],
+                                                     t[1]))
+                duplicates.append((name, lineage))
+
+    # Print results
+    print('{0} species-level matches ({1:.1f}%)'.format(
+        species_match, species_match/count*100))
+    print('{0} genus-level matches ({1:.1f}%)'.format(genus_match,
+                                                      genus_match/count*100))
+    if family_match > 0:
+        print('{0} family-level matches ({1:.1f}%)'.format(
+            family_match, family_match/count*100))
+    if no_match > 0:
+        print('{0} FAILURES ({1:.1f}%)'.format(no_match, no_match/count*100))
+
+    if len(duplicates) > 0:
+        print('\n{0} duplicates:'.format(len(duplicates)))
+        for dup in duplicates:
+            print('{0}\t{1}'.format(dup[0], dup[1]))
+
+    # Write to file
+    if not exists(destination_dir):
+        makedirs(destination_dir)
+
+    with open(join(destination_dir, 'expected-taxonomy.tsv'), "w") as dest:
```

Comment:
This will overwrite the file if it already exists - is that what you want to do? There's not a right or wrong way to handle this, it's just worth thinking about whether you would want to error if the file already exists to avoid losing work if the user doesn't realize they're about to overwrite a file, or if you just want to overwrite. This decision generally depends on how hard it would be to regenerate the overwritten file if the user didn't really want to overwrite it.